src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SegmentFilters.cs

```diff
@@ -128,5 +128,37 @@ public static bool HasSegments(Hl7v2Data hl7v2Data, string segmentIdContent)
 
             return result;
         }
+
+        public static List<Hl7v2Data> SplitDataBySegments(Hl7v2Data hl7v2Data, string segmentIdSeparators)
+        {
+            var results = new List<Hl7v2Data>();
+            var result = new Hl7v2Data();
+            var segmentIds = segmentIdSeparators.Split(@"|", StringSplitOptions.RemoveEmptyEntries);
+
+            if (segmentIdSeparators == string.Empty || segmentIds.Intersect(hl7v2Data.Meta).ToList().Count == 0)
+            {
+                results.Add(hl7v2Data);
+                return results;
+            }
+
+            for (var i = 0; i < hl7v2Data.Meta.Count; ++i)
+            {
+                if (segmentIds.Contains(hl7v2Data.Meta[i]))
```

Comment:
try to use hashset instead of array of string?

Reply:
Updated it. It would be better.